addons/account/models/account_account.py

```diff
@@ -534,10 +541,18 @@ def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None
         """, [partner_id, company_id])
         return [r[0] for r in self._cr.fetchall()]
 
+    @api.model
+    def _get_most_used_account_for_partner(self, company_id, partner_id, move_type=None):
+        return self._get_most_used_accounts_for_partner(company_id, partner_id, move_type=move_type, join='inner join', limit=1)
```

Comment:
This shouldn't return a list since you are asking for only one value
It isn't `_compute_account_id`'s burden to check for the existence of the account and parse the result